packages/katello/katello/hostname-change.rb

```diff
@@ -220,12 +220,15 @@ def using_custom_certs?(scenario_answers)
       scenario_answers["certs"]["server_key"]
     end
 
-    def delete_puppet_certs
+    def regenerate_puppet_certs
       return unless @scenario_answers['puppet'].is_a?(Hash)
 
       puppet_ssldir = @scenario_answers['puppet']['ssldir']
 
-      run_cmd("rm -rf '#{puppet_ssldir}'")
+      return unless puppet_ssldir
+
+      run_cmd("rm -rf '#{puppet_ssldir}'/*/#{@old_hostname}.*")
```

Comment:
Probably, yes. Need to test this tomorrow ;)

Reply:
```
Fatal error when running action 'clean'
  Error: Failed connecting to https://centos9-stream-katello-nightly.tanso.example.com:8140/puppet-ca/v1/certificate_status/
  Root cause: Failed to open TCP connection to centos9-stream-katello-nightly.tanso.example.com:8140 (Connection refused - connect(2) for "centos9-stream-katello-nightly.tanso.example.com" port 8140)
```

:shrug:

Reply:
```console
$ puppet ssl clean --help
...
* clean:
  Remove the private key and certificate related files for this host. If
  `--localca` is specified, then also remove this host's local copy of the
  CA certificate(s) and CRL bundle. if `--target CERTNAME` is specified, then
  remove the files for the specified device on this host instead of this host.
...
```

So perhaps `puppet ssl clean --target '#{@old_hostname}'` works?

Reply:
The "Failed connecting" is the `puppetserver ca clean` line. :/

Reply:
What does work is `puppetserver ca delete --certname …`, but that only deletes `/etc/puppetlabs/puppetserver/ca/signed/centos9-stream-katello-nightly.tanso.example.com.pem`, which is not helping much.

OTOH, it doesn't hurt if the old certs are there (unless you want to clean up traces of the old name, which some people do)

Reply:
I think I'd prefer to just merge the current state: it works and fixes a bug that we exposed by the change in puppet-foreman_proxy.

We can then go on and see if a more complicated puppet CA dance would be nicer, but not this year.

Reply:
And if you just use `puppet ssl clean --target '#{@old_hostname}'` instead of the `rm -rf`?